mbuild/conversion.py

```diff
@@ -1147,14 +1156,44 @@ def to_parmed(
     --------
     parmed.structure.Structure : Details on the ParmEd Structure object
     """
+
+    def _pull_hierarchical_attrs(
```

Comment:
Regarding the failing test, it's the `test_resname_parmed` in `test_compound.py`, specifically, when we try convert a compound (with not top level box) to parmed, so the `infer_hierarchy` step stops short at determining the lower level residues. A quick fix for this will be to add a line to define a box for that compound/system.

Reply:
This looks great! I actually made many changes to this branch already too but haven't gotten back to it either. I'll wrap up my changes and push them soon. I think it would be easier to brain storm at that point.